lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -111,25 +120,81 @@ export default function FilePickerApp({ onSubmit }) {
    * render.
    */
   const [shouldSubmit, setShouldSubmit] = useState(false);
-  const submit = useCallback(() => setShouldSubmit(true), []);
+
+  const [deepLinkingFields, setDeepLinkingFields] = useState(
+    /** @type {Record<string,string>|null} */ (null)
+  );
+
+  const submit = useCallback(
+    /** @param {Content} content */
+    async content => {
+      async function getDeepLinkingFields() {
```

Comment:
This callback can potentially streamlined a bit:

- The `content` argument is non-nullable, so you don't need to check if is truthy
- If the content of `getDeepLinkingFields` is inlined where it is used, that avoids needing two checks for whether `deepLinkingAPI` is set, since the surrounding `if` block has a check
- Since the `if (deepLinkingAPI ...)` is at the end of the function with no code after it, you can invert the logic and make it an early bail-out, reducing the number of levels of indentation and avoiding the `return` in the `catch` clause

A sketch:

```js
async content => {
  // Set shouldSubmit to true early to show the spinner while fetching form fields
  setShouldSubmit(true);
                                                                                   
  if (!deepLinkingAPI || deepLinkingFields) {
    return;
  }
                                                                                   
  // When deepLinkingAPI is present we want to call the backend to return the form
  // fields we'll forward to the LMS to complete the Deep Linking request
  try {
    const data = {
      ...deepLinkingAPI.data,
      content,
      extra_params: {
        groupSet: groupConfig.useGroupSet ? groupConfig.groupSet : null,
      },
    };
    setDeepLinkingFields(
      await apiCall({
        authToken: authToken,
        path: deepLinkingAPI.path,
        data,
      })
    );
  } catch (error) {
    setErrorInfo({
      message: 'Unable to configure assignment',
      error: error,
    });
    setShouldSubmit(false);
    setContent(null);
  }
},
```